### ISSUE TITLE
Unwrap Lists of objects

### ISSUE THREAD

**Issue by MaksimOrlov** [springfox/springfox#1349]
-  Version 2.5.0. 

 -  Feature Request. 

Good afternoon.

I was faced with a situation, when I have to describe parameters for the complex request and all parameters formatted as "text/plain" (for example products[0].shiftProducts[1].state=3).

For now, sprinfox when faced parameter like List&lt;MyObject&gt; generates type array[].

Where 

```java
MyObject {

private List<MyObject2>

getter 

setters
}

MyObject2 {

Integer param1;

String param2;

getters

setters

}
```

My controller looks like that 

```java
@ApiOperation(value = "Creating new order. Manager endpoint.",
		    notes = "This method creates a new order. Sends all notifications.",
		    response = NewOrders.class,
		    produces = "application/json")
	@PreAuthorize("hasRole('ROLE_MANAGER')")
	@RequestMapping(method=RequestMethod.POST, value="/addorder", params="mid")
	public @ResponseBody Object newOrder(@ModelAttribute @Validated({Default.class, ManagerCreate.class}) NewOrderWithShifts newOrder)
	{
		List<OrderWithShifts> newOrders = this.ordersService.addOrder(newOrder);
		
		return new NewOrders(newOrders);	
			
	}
```

I suggest to unwrap List generics and dump parameters in this way 
![git](https://cloud.githubusercontent.com/assets/7836239/16160811/c0e1f7b2-34d2-11e6-97f7-cf19bd194d4c.jpg)

I know, that related questions has been discussed :

#1209 
#988 

I made some "dirty" code changes, would you mind checking them?

https://github.com/MaksimOrlov/springfox/commit/37f95a6bc4d54903b92b6018d725fd82523d99a7

Thanks in advance.

**Comment by dilipkrish** [comment]
Without looking at your changes, there is already [a feature to edit this in swagger-ui](https://github.com/swagger-api/swagger-ui/pull/1589). This is an editor/viewer concern and I don't think we should change the rendering of the schema to make that easier.

**Comment by MaksimOrlov** [comment]
My suggestion was about parameters that is annotated with @ModelAttribute annotation.

The ModelAttributeParameterExpander in the 2.5.0 version of the Springfox library doesn't expand Lists with complex type. Because of this code at 134 line:

```
private Predicate<ModelAttributeField> simpleType() {
    return or(
            and(belongsToJavaPackage(),
              or(not(isCollection()), isMap())),
            isEnum());
```

It treats such parameters as simple typed parameters.

In swager-ui I can see next:

![git2](https://cloud.githubusercontent.com/assets/7836239/16162755/0701921a-34de-11e6-888c-dfd3ffd481c5.jpg)

Springfox generated this: 
```
{
"name":"delivery",
"in":"query",
"required":false,
"type":"array",
"items":{"$ref":"#/definitions/"},
"collectionFormat":"multi"
}
```

So, "items":{"$ref":"#/definitions/"} hide all parameters that were in delivery object.

```
public class NewOrderWithShifts extends OrderWithShifts {
	
	@Valid
	@Size(max=50)
	private List<DeliveryCost> delivery = new ArrayList<DeliveryCost>();
  
	public List<DeliveryCost> getDelivery() {
		return delivery;
	}

	public void setDelivery(List<DeliveryCost> delivery) {
		this.delivery = delivery;
	}

}
```

```
public class DeliveryCost {
	 
	@NotNull
	@Size(max=1500)
	@SafeHtml
	private String shopName;
	
	@NotNull
	@Min(0)
	@Digits(fraction = 2, integer = 12)
	private BigDecimal cost;

	public String getShopName() {
		return shopName;
	}

	public void setShopName(String shopName) {
		this.shopName = shopName;
	}

	public BigDecimal getCost() {
		return cost;
	}

	public void setCost(BigDecimal cost) {
		this.cost = cost;
	}
	
}
```

It shows just **delivery** and type **array[]**. But real parameters were **shopName**  and  **cost**. (**_delivery[].shopName_** and **_delivery[].cost_**)

In other words i suggest to replace LIst&lt;**_ComplexType_**&gt; to **_ComplexType_** and add '[]' to the name of the parameter.

**Comment by dilipkrish** [comment]
Got it! I dint pay attention sorry. To answer your question though its a known limitation. The support for model attributes itself is a stretch, its only meant to help with simple use cases. If your objects are this complex its a better idea to just make it a request body.

**Comment by MaksimOrlov** [comment]
I know that is a stretch.

But I'm not able to use @RequestBody annotation because it requires to send json or xml requests by client libraries, but for now my project is to old and to big to change format of requests. It will lead to overwhelming amount of changes at the all client libraries. 

But, I think, it's still conveniently to pass parameters as name-value key pairs. 

I would be grateful, if you consider my little enhancement, to check whether _List's generic_ is **Object** or not, and then expand it if so.

**Comment by dilipkrish** [comment]
@MaksimOrlov absolutely! Would you mind creating a PR with some tests as well. I'd be more than happy to take it in.

**Comment by MaksimOrlov** [comment]
Ok, I will create a PR with tests.

**Comment by MaksimOrlov** [comment]
Hi. I've created a PR https://github.com/springfox/springfox/pull/1350
